apps/src/code-studio/progressRedux.js

```diff
@@ -636,48 +635,50 @@ export function statusForLevel(level, levelProgress) {
 }
 
 /**
- * Groups lessons (aka stages) according to category.
+ * Groups lessons according to LessonGroup.
  * @returns {Object[]}
  * {string} Object.name
  * {string[]} Object.lessonNames
  * {Object[]} Object.stageLevels
  */
-export const categorizedLessons = (state, includeBonusLevels = false) => {
-  let byCategory = {};
+export const groupedLessons = (state, includeBonusLevels = false) => {
+  let byGroup = {};
 
   const allLevels = levelsByLesson(state);
 
-  state.stages.forEach((stage, index) => {
-    const category = stage.flex_category;
-    const lesson = lessonFromStageAtIndex(state, index);
-    let stageLevels = allLevels[index];
+  state.stages.forEach((lesson, index) => {
+    const group = lesson.lesson_group_name
```

Comment:
New lines 650 - 652 are really the key to this PR. This makes it so if we have a lesson group for a stage we group by that instead of the flex category. Otherwise we will continue to use the flex category. The rest of the changes here are just naming updates to move toward the world we want.